src/H5VLcallback.c

```diff
@@ -371,6 +951,68 @@ H5VL__common_optional_op(hid_t id, H5I_type_t id_type, H5VL_reg_opt_oper_t reg_o
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5VL__common_optional_op() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5VL__common_api_op
+ *
+ * Purpose:     Provide common wrapping for VOL callback API routines.
+ *
```

Comment:
That should explain much more about what it's really doing and why it needs to do that. That routine looks very much like a hack to me and it might be cleaning up some of the previous code, it is still as bad in my opinion. That's a lot of effort I feel to do something that is still not good in the end, even if it does make external wrappers "work".